airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -150,12 +162,45 @@ public static List<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair>
               jsonNodes.get(0).get("phase"));
           streamsUnderVacuuming.add(io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair.fromConfiguredAirbyteSteam(stream));
         }
-      } catch (SQLException e) {
+      } catch (final SQLException e) {
         // Assume it's safe to progress and skip relation node and vaccuum validation
         LOGGER.warn("Failed to fetch vacuum for table {} info. Going to move ahead with the sync assuming it's safe", fullTableName, e);
       }
     });
     return streamsUnderVacuuming;
   }
 
+  public static Map<AirbyteStreamNameNamespacePair, TableBlockSize> getTableBlockSizeForStream(final JdbcDatabase database,
```

Comment:
so page size/blocks size can be configured differently per table?

Reply:
It's a server setting in fact.
That block size is also used for some WAL logs definition, shared buffers and all sort of things. it's a low level setting on the server.
There is of course a body of literary work around how to optimize it - https://www.postgresql.org/docs/current/runtime-config-preset.html